crates/ty_python_semantic/src/module_resolver/resolver.rs

```diff
@@ -1006,6 +1030,32 @@ impl PackageKind {
     }
 }
 
+/// Info about the `py.typed` file for this package
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub(crate) enum PyTyped {
+    /// No `py.typed` was found
+    Untyped,
+    /// A `py.typed` was found containing "partial"
+    Partial,
+    /// A `py.typed` was found (not partial)
+    Full,
+}
+
+impl PyTyped {
+    /// Inherit py.typed info from the parent package
+    ///
+    /// > This marker applies recursively: if a top-level package includes it,
+    /// > all its sub-packages MUST support type checking as well.
+    ///
+    /// This implementation implies that once a `py.typed` is specified
+    /// all child packages inherit it, so they can never become Untyped.
+    /// However they can override whether that's Full or Partial by
+    /// redeclaring a `py.typed` file of their own.
+    fn inherit_parent(self, parent: Self) -> Self {
```

Comment:
No idea if this is correct but i like it conceptually

Reply:
The spec isn't totally clear here, it just says "This marker applies recursively: if a top-level package includes it, all its sub-packages MUST support type checking as well." It doesn't say anything about a sub-package going to `partial` where its parent was "full", or whatever. But I think this behavior is sensible.